### PR TITLE
Implement a rowid type.

### PR DESCRIPTION
**Pull request by fengttt**
Also, make TS an "supported" tae type.

## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [x] Code Refactoring

## Which issue(s) this PR fixes:

issue #
Cannot use decimal128 as rowid type.  Not all 16 bytes are valid decimal128 values,  different 16 bytes may represent
same decimal128 values.   Overall one cannot just construct a decimal128 and compare them.

## What this PR does / why we need it:
Create a new rowid type, [16]byte.

### CODE REVIEW

**Review by XuPeng-SH — APPROVED** [review]
lgtm

File: pkg/vm/engine/tae/mergesort/txnts/heap.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+// ordering for the Less method, so Push adds items while Pop removes the
+// highest-priority item from the queue. The Examples include such an
+// implementation; the file example_pq_test.go has the complete source.
+package txnts
+
+// Init establishes the heap invariants required by the other routines in this package.
+// Init is idempotent with respect to the heap invariants
+// and may be called whenever the heap invariants may have been invalidated.
+// The complexity is Operator(n) where n = len(h).
+func heapInit(h heapSlice) {
```
Comment (daviszhen) [inline pkg/vm/engine/tae/mergesort/txnts/heap.go:22]: The heap here is almost the same as the one in mergesort/rowid.

File: pkg/container/types/types.go [hunk 2]
```diff
@@ -70,12 +70,18 @@ const (
...
+	T_TS    T = 100
+	T_Rowid T = 101
 
 	// system family
 	T_tuple T = 201
 )
 
+const (
+	TxnTsSize = 12
+	RowidSize = 16
```
[the remark this answers is not in the document]
Comment (fengttt) [reply pkg/container/types/types.go:82]: That should be easy.   However, why 16 bytes is not enough for rowid?   That is 2^128, a very large number.

File: pkg/vm/engine/tae/mergesort/txnts/heap.go (hunk 1 above)
Comment (daviszhen) [reply pkg/vm/engine/tae/mergesort/txnts/heap.go:22]: The golang container package has the heap too

**Pull request merged**